lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -8,10 +8,25 @@ module BuiltIn
       class RaiseError
         include Composable
 
-        def initialize(expected_error_or_message=Exception, expected_message=nil, &block)
+        # @api public
+        # Configures the bare `raise_error` warning that would otherwise be
+        # issued when using the matcher without a specific error or message.
+        def self.allow_bare_raise_error=(value)
+          @allow_bare_raise_error = value
+        end
+
+        # @private
+        def self.allow_bare_raise_error?
+          @allow_bare_raise_error ||= false
+        end
```

Comment:
We have all our configuration exposed off of `RSpec::Expectations.configuration`, so it seems odd to have this other config here.